### PR TITLE
Feat: step function trigger lambda

### PR DESCRIPTION
**Pull request by kishore03109**

### CODE REVIEW

File: .env-example [hunk 1]
```diff
@@ -44,7 +44,9 @@ export POSTMAN_API_KEY=""
 export POSTMAN_SMS_CRED_NAME=""
 
 # Amplify
-AWS_REGION="ap-southeast-1"
+export AWS_REGION="ap-southeast-1"
+export AWS_ACCOUNT_NUMBER="000000000000"
+export IS_SITE_LAUNCH_DEV=true
```
Comment (prestonlimlianjie) [inline .env-example:49]: We should use the existing `NODE_DEV` env var for this

File: serverless.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ provider:
     INCOMING_QUEUE_URL: ${env:INCOMING_QUEUE_URL}
     OUTGOING_QUEUE_URL: ${env:OUTGOING_QUEUE_URL}
     AWS_REGION: ${env:AWS_REGION}
+    IS_PROD: ${env:IS_SITE_LAUNCH_DEV}
```
Comment (prestonlimlianjie) [inline serverless.yml:20]: I was looking through the lambda functions to see if `IS_PROD` is being used anywhere and wasn't able to find it. Is this still necessary?

Comment (kishore03109) [reply serverless.yml]: opps: this was a typo, fixed [here]([4abf7bb](https://github.com/isomerpages/isomercms-backend/pull/538/commits/4abf7bbf969779ded606fdbacaff4e6fe62bb08a))

File: .env-example (hunk 1 above)
Comment (kishore03109) [reply .env-example:49]: [4abf7bb](https://github.com/isomerpages/isomercms-backend/pull/538/commits/4abf7bbf969779ded606fdbacaff4e6fe62bb08a)

File: microservices/site-launch/lambda-functions/step-functions-trigger/index.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          : `https://lambda.${AWS_REGION}.amazonaws.com`,
+    })
+    const params: Lambda.Types.InvocationRequest = {
+      FunctionName: `isomercms-dev-failureNotification`,
+      InvocationType: "RequestResponse",
+      Payload: JSON.stringify(event),
+    }
+    lambda.invoke(params, (res, error) => {
+      console.log(res, error)
+    })
```
Comment (seaerchin) [inline microservices/site-launch/lambda-functions/step-functions-trigger/index.ts:42]: do we actually need to do this? it seems like the yml definition does this automatically for us as the `OutputPath: $.Payload` is also triggered on function error

File: microservices/site-launch/lambda-functions/step-functions-trigger/index.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    const stateMachineArn = `arn:aws:states:${AWS_REGION}:${AWS_ACCOUNT_NUMBER}:stateMachine:siteLaunch`
+
+    const params = {
+      stateMachineArn,
+      input: JSON.stringify(event),
+    }
+
+    stepFunctions.startExecution(params, (res, err) => {
+      console.log(`Your state machine ${stateMachineArn} executed successfully`)
```
Comment (seaerchin) [inline microservices/site-launch/lambda-functions/step-functions-trigger/index.ts:22]: should be logger

File: microservices/site-launch/lambda-functions/step-functions-trigger/index.ts (hunk 3 above)
Comment (kishore03109) [reply microservices/site-launch/lambda-functions/step-functions-trigger/index.ts:42]: yup, the `site-launch.asl.yaml` is to define how the shape of the message is for lambdas **_within_** the step functions, which this particular lambda does not reside in!

File: microservices/site-launch/lambda-functions/step-functions-trigger/index.ts (hunk 4 above)
Comment (kishore03109) [reply microservices/site-launch/lambda-functions/step-functions-trigger/index.ts:22]: d500020

Comment (kishore03109) [reply microservices/site-launch/lambda-functions/step-functions-trigger/index.ts]: 21a9853

File: microservices/site-launch/lambda-functions/step-functions-trigger/index.ts (hunk 3 above)
Comment (seaerchin) [reply microservices/site-launch/lambda-functions/step-functions-trigger/index.ts]: ah gotcha! thanks for clarifying!

**Review by seaerchin — APPROVED** [review]

**Pull request merged**